planner/core/find_best_task.go

```diff
@@ -1495,6 +1495,11 @@ func (ds *DataSource) convertToTableScan(prop *property.PhysicalProperty, candid
 			// If ts is a single partition, then this partition table is in static-only prune, then we should not choose mpp execution.
 			return &mppTask{}, nil
 		}
+		origColsLen := len(ts.Columns)
+		ts.Columns = ExpandVirtualColumn(ts.Columns, ts.schema, ts.Table.Columns)
```

Comment:
If the virtual column is built on constant, then len(ts.Columns) == origColsLen.
Is it expected to build the MPP in this case?

Reply:
Nice catch.